detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenMethodCall.kt

```diff
@@ -45,6 +46,14 @@ class ForbiddenMethodCall(config: Config = Config.empty) : Rule(config) {
     )
 
     private val forbiddenMethods = valueOrDefaultCommaSeparated(METHODS, DEFAULT_METHODS)
+        .map { methodSignature ->
+            val groups = METHOD_PARAM_REGEX.find(methodSignature)?.groups
+            val methodName = groups?.get(METHOD_GROUP_NAME)?.value
+            val params = groups?.get(METHOD_PARAM_GROUP_NAME)?.value?.split(",")
+                ?.map { it.trim() }
+                ?.filter { it.isNotBlank() }
```

Comment:
Please replace with something like:

```suggestion
            val groups = methodSignature.split('(',')')
            val methodName = groups[0].trim()
            val params = if (groups.size <= 1) {
                null
            } else {
                groups[1].split(",")
                    .map(String::trim)
                    .filter(String::isNotBlank)
            }
```

Regexes are hard to maintain in the long run, especially for shared codebases like Detekt.

Reply:
If you wish, you can also create a util function inside the `detekt-psi-utils` package and write a unit test (like a `.parseFqNameAndParams()` or so).

Reply:
Probably, because it's not that easy because some functions can be called `` `like, this`(Int)``.

And, if we want to move out this code,  it should be placed in tooling utils instead of psi utils, right? This code is to parse the configuration. It's not related with psi.

Reply:
> it should be placed in tooling utils instead of psi utils, right?

I checked and `detekt-psi-utils` is already imported inside the `-rules` modules as an `api` dependency. The `detekt-tooling` is not and I don't think it's worth to add a module dependency for a function.

Not to be done in this PR, but I think we should rename `detekt-psi-utils` to `detekt-rules-utils` (given that inside there is also not PSI related utils) or create a separate `detekt-rules-utils` just for those functions.

Reply:
Good idea. My first thought was to use regex and I was so fixed on this approach that I missed simpler one.